src/views/ApartmentApp/components/StudentApplication/components/HallSelection/components/HallListItem/index.js

```diff
@@ -102,7 +94,7 @@ const HallListItem = ({
               <InputLabel>Rank</InputLabel>
               <Select
                 disabled={disabled}
-                value={hallRankValue}
+                value={hallRank}
```

Comment:
Does that actually work, to use the prop directly? I thought that didn't work when I tried testing it in the past

Reply:
I've been using it all day during my testing and it worked perfectly.

Reply:
In case it isn't obvious why that works, I guess I should clarify. The real source of truth for the hall choices are stored as state on the StudentApplication component. When the user changes a hall rank or name, you call that component's method to handle changes and changing that component's state accordingly. This state change results in new props cascading through the HallSelection component and down to this component. The new props cause the component to re-render, and thus the displayed values update accordingly.

I'm actually not totally clear on why your previous way was working. Either the new props were re-calculating the state every time, or else I would expect that the component's state never updated at all, since you never called the setState functions after the initial useEffect.

Reply:
The previous way worked because having the props explicitly listed in the dependency array tells the useEffect to re-run every time those props get changed.

I think the real reason I thought I needed that was because of a problem in the `handleHallInputChange` in the `StudentApplication` component, where I was using the setState things incorrectly and I thought the problem that in the list items not at the top of the component tree. I tried many ridiculous variation in my attempts to fix that bug, which obvious did not succeed because I didn't realize that problem was in `StudentApplication` and the setState things behaved differently than how I thought they did at the time

Reply:
I also think I used to have the state variable in this component so that the `handleNameInputChange` and such could do a `setState` for the thing that the user selected, and then after that I added in the functionality of having it mapped to the state array from higher up